packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -389,12 +396,27 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
     _hideOptionsAction.enabled = enabled;
   }
 
+  void _updateActions() {
+    _setActionsEnabled(_focusNode.hasFocus && _selection == null && _options.isNotEmpty);
+  }
+
+  bool _floatingOptionsUpdateScheduled = false;
   // Hide or show the options overlay, if needed.
   void _updateOverlay() {
-    _setActionsEnabled(_focusNode.hasFocus && _selection == null && _options.isNotEmpty);
+    if (SchedulerBinding.instance.schedulerPhase == SchedulerPhase.persistentCallbacks) {
```

Comment:
So the problem was that _updateOverlay was called on a tap callback, in the persistentCallbacks phase, and so the OverlayEntry was updated for the previous frame?

Reply:
No tap callbacks should never be called in persistent callbacks.
The problem was if the text isn't changed by the tap, it's not going to call `_updateOverlay` so that `shouldShowOptions` gets re-evaluated. 
This is just a small change to make sure the caller of `_updateOverlay` doesn't have to know whether it's safe to do an overlay update right away or it has to schedule an update.

Reply:
Got it, sounds good.